cli/src/pcluster/resources/imagebuilder/update_and_reboot.yaml

```diff
@@ -107,6 +107,22 @@ phases:
                 update-grub
               fi
 
+      - name: WaitCloudInitCompletion
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              PLATFORM='{{ build.PlatformName.outputs.stdout }}'
+              
+              # Wait for completion of os updates performed by cloud-init
+              if [[ ${!PLATFORM} == DEBIAN ]]; then
+                while [ ! -f /var/lib/cloud/instance/boot-finished ]; do echo 'Waiting for cloud-init...'; sleep 1; done
+                flock $(apt-config shell StateDir Dir::State/d | sed -r "s/.*'(.*)\/?'$/\1/")/daily_lock systemctl disable --now apt-daily.timer apt-daily.service apt-daily-upgrade.timer apt-daily-upgrade.service
+                sed "/Update-Package-Lists/s/\"1\"/\"0\"/; /Unattended-Upgrade/s/\"1\"/\"0\"/;" /etc/apt/apt.conf.d/20auto-upgrades > "/etc/apt/apt.conf.d/51pcluster-unattended-upgrades"
+                DEBIAN_FRONTEND=noninteractive apt-get -y update
+              fi
```

Comment:
Could we put some comments explaining what we are doing here and why? Even in a later PR.

Reply:
Oh, I see this was already present a bit below in the code.

Reply:
Adding a comment.